lib/dependenciesInstall.js

```diff
@@ -1,40 +1,71 @@
+var fs = require('fs');
+var path = require('path');
+
+// This function gets called after every plugin addition, as specified in plugin.xml
 module.exports = function(context) {
+
+    var options = context.opts;
     var Q = context.requireCordovaModule('q');
-    var npm = context.requireCordovaModule('npm');
-    var fs = require('fs');
-    var path = require('path');
 
     // package.json file is a sister folder to 'lib' folder, in which we currently are executing this script from
-    var packageJSONDir = path.join(__dirname, '../'); 
+    var packageJSONDir = path.join(__dirname, '../');
     var packageJSONLocation = path.join(packageJSONDir, 'package.json');
-    var dependencies = JSON.parse(fs.readFileSync(packageJSONLocation, 'utf-8')).dependencies;
-    
+    var packageJSON = JSON.parse(fs.readFileSync(packageJSONLocation, 'utf-8'));
+
+    var pluginName = packageJSON.name;
```

Comment:
Isn't `pluginName ` a constant?

If this can be factored into constant then you could move the `if ` statement below closer to beginning of hook so you don't need to read file and parse it if the plugin already installed, just exit immediately.